packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -507,13 +507,11 @@ function __vite__mapDeps(indexes) {
   return indexes.map((i) => __vite__mapDeps.viteFileDeps[i])
 }\n`
 
-          // inject extra code before sourcemap comment
-          const mapFileCommentMatch =
-            convertSourceMap.mapFileCommentRegex.exec(code)
-          if (mapFileCommentMatch) {
-            s.appendRight(mapFileCommentMatch.index, mapDepsCode)
+          // inject extra code at the top or next line of hashbang
+          if (code.startsWith('#!')) {
+            s.prependLeft(code.indexOf('\n') + 1, mapDepsCode)
           } else {
-            s.append(mapDepsCode)
+            s.prepend(mapDepsCode)
```

Comment:
if `fileDeps.length == 0`, maybe we do not need exec  `s.prependLeft` or `prepend`

because sometimes `__vite__mapDeps` is not used

![image](https://github.com/vitejs/vite/assets/38517192/8a642ff8-e596-47f0-b466-51cc0ccdb797)

Reply:
We can use `[]` instead of `__vite__mapDeps([${renderedDeps.join(',')}])` if `renderedDeps.length` is 0. Then, we can skip injecting `__vite__mapDeps`. Contributions are welcome 👍